buildtest/cli/report.py

```diff
@@ -602,12 +602,13 @@ def get_buildspecs(self):
 
     def get_failed_tests(self):
         """Return a list of failed test names from report file"""
-        test_names = []
+        test_names = set()
         for buildspec in self.filtered_buildspecs:
             for name in self.report[buildspec].keys():
-                if self.report[buildspec][name][0]["state"] == "FAIL":
-                    test_names.append(name)
-        return test_names
+                for trial in self.report[buildspec][name]:
+                    if trial["state"] == "FAIL":
+                        test_names.add(name)
```

Comment:
if i understand correctly we want to break out of this for loop when we get the first instance of test that has a `FAIL` instead of going through the entire list of test runs for a given test lets say `hello-world` which has 5 test. With this for loop we will iterate over all 5 runs.

I think you can use `break` to get out of this loop which would work.